### PR TITLE
Attempt to move room aliases on room upgrades

### PR DESCRIPTION
**Pull request by richvdh**
Builds on #515

### CODE REVIEW

File: synapse/handlers/room.py [hunk 1]
```diff
@@ -294,7 +294,112 @@ def clone_exiting_room(
...
+
+        try:
+            # this can fail if, for some reason, our user doesn't have perms to send
+            # m.room.aliases events in the old room (note that we've already checked that
+            # they have perms to send a tombstone event, so that's not terribly likely).
+            #
+            # If that happens, it's regrettable, but we should carry on: it's the same
+            # as when you remove an alias from the directory normally - it just means that
+            # the aliases event gets out of sync with the directory
+            # (cf https://github.com/vector-im/riot-web/issues/2369)
```
Comment (erikjohnston) [inline synapse/handlers/room.py]: I'd be reasonably happy special casing upgrading of rooms to always allow moving aliases on room upgrade tbh. (Though not suggesting this should be done in this PR)

Comment (richvdh) [reply synapse/handlers/room.py]: We *do* move the alias anyway in this case: it's just that we can't send the `m.room.aliases` event.

Comment (erikjohnston) [reply synapse/handlers/room.py]: Oh, my bad.

**Comment by richvdh** [comment]
(this is also now failing due to #4099 landing)

File: synapse/handlers/room.py [hunk 2]
```diff
@@ -294,7 +294,112 @@ def clone_exiting_room(
...
+                yield directory_handler.delete_association(
+                    requester, alias, send_event=False,
+                )
+            except SynapseError as e:
+                logger.warning(
+                    "Unable to remove alias %s from old room: %s",
+                    alias, e,
+                )
+            else:
+                removed_aliases.append(alias_str)
```
[the remark this answers is not in the document]
Comment (JonLan2002) [reply synapse/handlers/room.py]: else is "if there was no exception"

**Pull request merged**